proposals/0024-repair-and-restart.md

```diff
@@ -0,0 +1,302 @@
+---
+simd: '0024'
+title: Optimistic cluster restart automation
+authors:
+  - Wen Xu (Solana Labs)
+category: Standard
+type: Core
+status: Draft
+created: 2023-04-07
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+During a cluster restart following an outage, use gossip to exchange local
+status and automatically reach consensus on the block to restart from. Proceed
+to restart if validators in the restart can reach agreement, or print debug
+information and halt otherwise.
+
+## New Terminology
+
+* "cluster restart": When there is an outage such that the whole cluster
+stalls, human may need to restart most of the validators with a sane state so
+that the cluster can continue to function. This is different from sporadic
+single validator restart which does not impact the cluster. See
+[cluster restart](https://docs.solana.com/running-validator/restart-cluster)
+for details.
+
+* "optimistically confirmed block": a block which gets the votes from the
+majority of the validators in a cluster (> 2/3 stake). Our algorithm tries to
+guarantee that an optimistically confirmed will never be rolled back. When we 
+are performing cluster restart, we normally start from the highest 
+optimistically confirmed block, but it's also okay to start from a child of the 
+highest optimistically confirmed block as long as consensus can be reached.
+
```

Comment:
😩 how are these not in https://docs.solana.com/terminology yet! Would you mind sending a PR into the monorepo to add these terms, they really aren't new for this SIMD!

Reply:
Sure, but can I keep these here so the document is self-contained?

Reply:
We can get those terms on docs.solana.com pretty quickly too, but meanwhile sure makes sense!